File: apps/go/domain/type_/fizz_buzz_type_01.go
package type_

import (
	"strconv"

	"github.com/k2works/getting-started-tdd/apps/go/domain/model"
)

// FizzBuzzType01 は通常の FizzBuzz を生成します。
type FizzBuzzType01 struct{ fizzBuzzTypeBase }

func (f FizzBuzzType01) Generate(number int) model.FizzBuzzValue {
	if f.isFizz(number) && f.isBuzz(number) {
		return model.NewFizzBuzzValue(number, "FizzBuzz")
	}
	if f.isFizz(number) {
		return model.NewFizzBuzzValue(number, "Fizz")
	}
	if f.isBuzz(number) {
		return model.NewFizzBuzzValue(number, "Buzz")
	}
	return model.NewFizzBuzzValue(number, strconv.Itoa(number))
}

File: apps/java/src/main/java/tdd/fizzbuzz/FizzBuzzType01.java
package tdd.fizzbuzz;

public class FizzBuzzType01 extends FizzBuzzType {

    @Override
    public String generate(int number) {
        if (isFizzBuzz(number)) {
            return "FizzBuzz";
        }
        if (isFizz(number)) {
            return "Fizz";
        }
        if (isBuzz(number)) {
            return "Buzz";
        }
        return Integer.toString(number);
    }
}
